hedera-node/src/main/java/com/hedera/services/utils/SignedTxnAccessor.java

```diff
@@ -98,6 +111,14 @@ public SignedTxnAccessor(byte[] signedTxnWrapperBytes) throws InvalidProtocolBuf
 		numSigPairs = sigMap.getSigPairCount();
 		utf8MemoBytes = StringUtils.getBytesUtf8(memo);
 		memoHasZeroByte = Arrays.contains(utf8MemoBytes, (byte) 0);
+
+		getFunction();
```

Comment:
Does this line serve any purpose?

Reply:
It sets the `function` field so we can decide next whether to compute the `CryptoTransfer` or `ConsensusSubmitMessage` usage metadata.

Reply:
Yeah. I was thinking it should be `function = getFunction(); ` , but `getFunction();` here already serves the purpose.